File: TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/dialog/DateSpinnerAdapter.java
package com.skedgo.tripkit.ui.dialog;

import android.content.Context;
import android.widget.ArrayAdapter;

import java.util.List;

public class DateSpinnerAdapter extends ArrayAdapter<String> {
  private List<String> dates;

  public DateSpinnerAdapter(Context context, int resource, List<String> dates) {
    super(context, resource, dates);
    this.dates = dates;
  }

  public void setDates(List<String> dates) {
    this.dates.clear();
    this.dates.addAll(dates);
    this.notifyDataSetChanged();
  }

  @Override
  public String getItem(int position) {
    return dates.get(position);
  }
}

File: TripKitAndroidUI/src/main/java/com/skedgo/tripkit/ui/dialog/DateSpinnerAdapter.kt
package com.skedgo.tripkit.ui.dialog

import android.content.Context
import android.widget.ArrayAdapter

class DateSpinnerAdapter(
    context: Context?,
    resource: Int,
    private val dates: MutableList<String>
) : ArrayAdapter<String>(
    context!!, resource, dates
) {
    fun setDates(dates: List<String>) {
        this.dates.clear()
        this.dates.addAll(dates)
        this.notifyDataSetChanged()
    }

    override fun getItem(position: Int): String {
        return dates[position]
    }
}
